js/reducers/passageReducer.js

```diff
@@ -100,4 +101,14 @@ function passageReducer(state = initialState, action) {
   }
 }
 
-export default passageReducer;
+function stateSavingPassageReducer(state, action) {
+  if (!state) {
+    state = assignToEmpty(defaultState, restoreState());
```

Comment:
Looking at some the Redux recipes in the documentation http://redux.js.org/docs/recipes/ServerRendering.html (Next Steps section) and http://redux.js.org/docs/advanced/AsyncActions.html

It seems like one way to solve this would be to have a `fetchData()` for the React-Router route or to use an async action to fetch the data from the cookie.

Thinking out loud here, but maybe what we want to implement on the reducer is a `RESTORE_SAVED_STATE` action which can pull in the state from the cookie and then the action would get called by the router on load.